internal/cloudinfo/machine_types.go

```diff
@@ -54,14 +56,38 @@ type VMKey struct {
 	cloud        string
 	service      string
 	region       string
-	location     string
 	instanceType string
 }
 
+type InstanceTypeMap struct {
+	sync.RWMutex
```

Comment:
Do not embed synchronization structures, it makes them publicly available.

Reply:
fixed